src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -114,6 +114,25 @@ public void executeShell(OutputStream output, String command, String... args) th
         }
     }
 
+    /** Execute a shell command with raw binary output.
+     *
+     * @param command main command to run, e.g. "screencap"
+     * @param args arguments to the command, e.g. "-p".
+     * @return combined stdout/stderr stream.
+     * @throws IOException
+     * @throws JadbException
+     */
+    public InputStream executeShellRaw(String command, String... args) throws IOException, JadbException {
+        Transport transport = getTransport();
+        StringBuilder shellLine = new StringBuilder(command);
+        for (String arg : args) {
```

Comment:
Could you please extract command line forming in separate private method?
And use it in both `executeShell` and `executeShellRaw`